charts/fleet/templates/deployment.yaml

```
@@ -1,12 +1,29 @@
-{{ $shards := list "" }}
-{{ if .Values.shards }}
-{{ $shards = concat $shards .Values.shards | uniq }}
-{{ end }}
-{{ range $shards }}
+{{- $shards := list (dict "id" "" "nodeSelector" dict) -}}
+{{- if .Values.shards -}}
+  {{- range .Values.shards -}}
+    {{- $shards = append $shards . -}}
+  {{- end -}}
+{{- end -}}
+
+{{- $uniqueShards := list -}}
+{{- range $shard := $shards -}}
+  {{- $found := false -}}
+  {{- range $uniqueShard := $uniqueShards -}}
+    {{- if eq $shard.id $uniqueShard.id -}}
+      {{- $found = true -}}
+      {{- break -}}
+    {{- end -}}
+  {{- end -}}
+  {{- if not $found -}}
+    {{- $uniqueShards = append $uniqueShards $shard -}}
+  {{- end -}}
+{{- end -}}
+
+{{ range $shard := $uniqueShards }}
```

This works, but there is a much simpler way ;)
```suggestion
{{ $shards = $shards | uniq }}

{{ range $shard := $shards }}
```

---

Already tried and it do not work because if two shards have the same id but not the same node selector it will not delete them, we just wan't the id to not be duplicated

---

This compare the whole list item

---

Yes, just updated it, thank you